core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaSchemaRegistry.java

```diff
@@ -1359,22 +1352,18 @@ public List<SubjectVersion> listVersionsForId(int id, String subject, boolean lo
     }
   }
 
-  private Set<String> extractUniqueSubjects(Iterator<SchemaRegistryKey> allKeys,
-                                            boolean returnDeletedSubjects)
-      throws StoreException {
-    Set<String> subjects = new LinkedHashSet<>();
-    while (allKeys.hasNext()) {
-      SchemaRegistryKey k = allKeys.next();
-      if (k instanceof SchemaKey) {
-        SchemaKey key = (SchemaKey) k;
-        SchemaValue value = (SchemaValue) kafkaStore.get(key);
-        if (value != null
-                && (!value.isDeleted() || returnDeletedSubjects)) {
-          subjects.add(key.getSubject());
-        }
-      }
+  private Set<String> extractUniqueSubjects(Iterator<SchemaRegistryValue> allVersions,
+                                            LookupFilter filter) {
+    Map<String, Boolean> subjects = new HashMap<>();
+    while (allVersions.hasNext()) {
+      SchemaValue value = (SchemaValue) allVersions.next();
+      subjects.merge(value.getSubject(), value.isDeleted(), (v1, v2) -> v1 && v2);
```

Comment:
Is the use of `&&` correct?  It seems that DEFAULT should return a subject if any version is not deleted, INCLUDE_DELETED should return a subject if any version exists, and DELETED_ONLY should return a subject if all versions are deleted.

Reply:
The `subjects` map is a mapping of <subject, all(...version.isDeleted)>
Therefore, the end result of the map will be the following two catagories:
- (subjects where all versions are deleted)                subA -> true 
- (subject where at least 1 version are non deleted) subB -> false 

And here's the  `shouldInclude()` return value for different param combination:
filter: DEFAULT, INCLUDE_DELETED, DELETED_ONLY
subA F,               T,                                T        
subB  T,                T,                                F

Reply:
I think this API is used to get the subject, so the filter is applying to the subject not schemas? 
ie. DEFAULT will return all non-deleted subjects, which is any subject that has any version exists.